packages/atomic/src/utils/set.ts

```diff
@@ -0,0 +1,10 @@
+// TODO: Replace by Set.prototype.intersection in July 2025
+export function intersection<T>(setA: Set<T>, setB: Set<T>): Set<T> {
```

Comment:
Where did you take the implementation?

Shouldn't we unit-test it if it's in-house?

Reply:
Self-made (well, copilot). could be worth testing indeed

Reply:
👎 the oneliner, while neater, has sub-optimal perf:
- First, the [...setA] goes through all values of setA once.
- Second, we go through all of them a second time with the filter function (and for each of them, we go through all the values of b)
-  Finally, we go through the 'result' once more to create the Set.

So, roughly `O(a+ab+n) ` (with n being the intersect length)

The long version is `O(ab)`.

To be noted, while this is true, it goes by the account the cost-per-singular operation is similar between the two methods.